### DISCUSSION TITLE
How are the edge_index in MNISTSuperpixel Dataset computed?

### DISCUSSION THREAD

**Discussion by domalbert** [pyg-team/pytorch_geometric#6026] [Q&A]
I was trying to use to transfer the MNIST dataset into superpixel representation using ```ToSlic``` and afterwards approximate the ```edge_index``` using ```KNNGraph```. 

```
import torch
import torch_geometric
from torchvision.datasets import MNIST
import torchvision.transforms as T
from torch_geometric.transforms import ToSLIC
from torch_geometric.transforms.knn_graph import KNNGraph
from torch_geometric.transforms.radius_graph import RadiusGraph

transform = T.Compose([T.ToTensor(),ToSLIC(n_segments=75),KNNGraph()])
dataset = MNIST('/tmp/MNIST', download=True, transform=transform)
train_set_x = dataset.data.numpy()#.reshape(1, 28,28, -1)
print(train_set_x.shape)
```

I visualized the graph representation for both MNIST and MNISTSuperpixel, however with strong differences in the result. For my custom adaption for MNIST I receive

![image](https://user-images.githubusercontent.com/90129268/202986544-22f48aa8-2540-42f0-a872-4ab1d38ead32.png)

while for inbuild Superpixel I obtain:

![image](https://user-images.githubusercontent.com/90129268/202987286-eea2436c-3831-4322-a019-eb2ad4763d49.png)
 
Can you give me guidance of how the edge_index is calculated within the MNISTSuperpixel representation?

If you need additional code let me know

**Comment by rusty1s** [comment]
I sadly don't have the SLIC parameters that were used to create the `MNISTSuperpixelDataset`. This dataset was pre-processed from the MoNet authors, I am sorry :(

**Reply by domalbert** [reply]
Okey. Thank you for your reply :-)
